homeassistant/components/sensor/google_travel_time.py

```diff
@@ -186,6 +215,65 @@ def update(self):
         elif atime is not None:
             options_copy['arrival_time'] = atime
 
-        self._matrix = self._client.distance_matrix(self._origin,
-                                                    self._destination,
-                                                    **options_copy)
+        # Convert device_trackers to google friendly location
+        if hasattr(self, '_origin_entity_id'):
+            self._origin = self._get_location_from_entity(
+                self._origin_entity_id
+            )
+
+        if hasattr(self, '_destination_entity_id'):
+            self._destination = self._get_location_from_entity(
+                self._destination_entity_id
+            )
+
+        if self._destination is not None and self._origin is not None:
+            self._matrix = self._client.distance_matrix(self._origin,
+                                                        self._destination,
+                                                        **options_copy)
+
+    def _get_location_from_entity(self, entity_id):
+        """Get the location from the entity state or attributes."""
+        entity = self._hass.states.get(entity_id)
+
+        if entity is None:
+            _LOGGER.error("Unable to find entity %s", entity_id)
+            self.valid_api_connection = False
+            return None
+
+        # Check if device is in a zone
```

Comment:
Consider using [`has_location`](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/location.py#L8).

```python
# Use these attributes instead of strings
from homeassistant.const import ATTR_LATITUDE, ATTR_LONGITUDE

if has_location(entity):
    return …
elif entity.state matches regex:
    return …
return None
```